### DISCUSSION TITLE
is there a way to get source/destination using the api?

### DISCUSSION THREAD

**Discussion by galyadid** [apache/airflow#21698] [Q&A]
Hey all, 

trying to understand if there's a way that the api can provide me the details of the source/destination of a dag.

for example, if I have a DAG that extract data from snowflake table, I want to be able to get this information from the api:
db name, schema name, table name

Do you know if it's possible? 

thanks!

**Comment by potiuk** [comment]
It is (for now) specific for the operator.  You can even get source code of a DAG so you can retrieve the information from that. But there is no "generic" schema/table/db information as Airflow tasks are very generic and not all of them (or even minority of them) are db/scheme/table related. This might change in the future with "lineage" integration but for now it's not available.
